File: Boj-Kotlin&Java/백준11048번이동하기.java
package 백준;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class 백준11048번이동하기 {
    static int n,m;
    static int[][] arr,dp;
    static StringTokenizer st;
    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        String[] nm=br.readLine().split(" ");
        n=Integer.parseInt(nm[0]);
        m=Integer.parseInt(nm[1]);
        arr=new int[n+1][m+1];
        dp=new int[n+1][m+1];
        for(int i=0; i<n; i++){
            st=new StringTokenizer(br.readLine());
            for(int j=0; j<m; j++)
                arr[i+1][j+1]=Integer.parseInt(st.nextToken());
        }
        int tmp=0;
        for(int i=1; i<=n; i++){
            for(int j=1; j<=m; j++){
                tmp=Math.max(Math.max(dp[i-1][j-1],dp[i-1][j]),dp[i][j-1]);
                dp[i][j]=tmp+arr[i][j];
            }
        }
        System.out.println(dp[n][m]);
    }
}

File: Kotlin/백준 11048번 이동하기.kt
import kotlin.math.max

fun main(args: Array<String>) {
    val br=System.`in`.bufferedReader()
    val (n,m)=br.readLine().split(" ").map {it.toInt()}
    val arr=Array(n+1){Array(m+1){0}}
    val dynamic=Array(n+1){Array(m+1){0}}
    for(i in 0 until n)
    {
        val input=br.readLine().split(" ").map { it.toInt() }
        for(j in input.indices)
            arr[i+1][j+1]=input[j]
    }

    var answer=0
    for(i in 1..n)
    {
        for(j in 1..m)
        {
            answer= max(max(dynamic[i-1][j-1],dynamic[i-1][j]),dynamic[i][j-1])
            dynamic[i][j]=answer+arr[i][j]
        }
    }
    println("${dynamic[n][m]}")

}